logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/asset/ScheduleNewAssetMessageUseCase.kt

```diff
@@ -115,18 +125,19 @@ internal class SendAssetMessageUseCaseImpl(
             )
 
             // We persist the asset message right away so that it can be displayed on the conversation screen loading
-            persistMessage(message).flatMap {
-                Either.Right(Unit)
-            }.onFailure {
-                kaliumLogger.e("Asset persist method failed")
-                updateAssetMessageUploadStatus(Message.UploadStatus.FAILED_UPLOAD, conversationId, message.id)
+            persistMessage(message).fold({
                 Either.Left(it)
-            }.flatMap {
-                uploadAssetAndUpdateMessage(message, conversationId)
-            }
+                                         }, {
+                // We schedule the asset upload and return the message ID
```

Comment:
Here it returns `Unit`, not message ID so it took me a few seconds to understand it 😅 maybe change it to be more like "We schedule the asset upload and return Either.Right(Unit) so later it's transformed to Either.Right(message.id)"? 

Also, maybe subtle change but I feel like this way it would look better 😄 
```
persistMessage(message)
    .onSuccess {
        scope.launch(dispatcher.io) { uploadAssetAndUpdateMessage(message, conversationId) }
    }
    .onFailure {
        updateAssetMessageUploadStatus(Message.UploadStatus.FAILED_UPLOAD, conversationId, message.id)
    }
```

Reply:
Fair enough. Thanks for making it look more readable 🤓